R/checkSummations.R

```diff
@@ -105,12 +112,26 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = NULL, sum
     # calculate differences for comparison
     comp <- comp %>%
       group_by(!!!syms(c("model", "scenario", "region", "period", "variable", "unit", "value"))) %>%
-      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor")), .groups = "drop") %>%
+      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor"), na.rm = TRUE),
+        summation = paste(
+          ifelse(
+            is.na(!!sym("child")),
+            "",
+            ifelse(
+              !!sym("factor") == 1,
+              paste0(!!sym("child"), " (", !!sym("childVal"), ")"),
+              paste0(!!sym("factor"), " * ", !!sym("child"), " (", !!sym("childVal"), ")")
+              )
+            ),
+          collapse = " + "),
+        .groups = "drop") %>%
       ungroup() %>%
       mutate(
         diff = !!sym("checkSum") - !!sym("value"),
-        reldiff = 100 * (!!sym("checkSum") - !!sym("value")) / !!sym("value")
-      )
+        reldiff = 100 * (!!sym("checkSum") - !!sym("value")) / !!sym("value"),
+        summation = gsub("\\+ \\-", "-", !!sym("summation"))
```

Comment:
Not sure if I understand the regexp correctly, but this replaces any occurrence of "+ -" by "-", right? Or do I misunderstand the meaning?

If this is the actual meaning, under which circumstances could you create the substring "+ - " in the concatenation you are performing?

Reply:
Whenever you have a negative factor, you would have a redundant plus sign in the equation string.
For example, without this code you could have 
`var A (0.01) + - 2 * var B (0.05)`
this regex just avoids this case, resulting in instead the below equation:
`var A (0.01) - 2 * var B (0.05)`